docs/decisions/0006-middleware-support-for-http-clients.rst

```diff
@@ -0,0 +1,40 @@
+middleware for HTTP client
+==========================================
+
+Status
+------
+
+
+
+Context
+-------
+
+We currently expose a HTTP client(axios instance) via ``getAuthenticatedHttpClient`` and ``getHttpClient`` used in
+our MFEs to make HTTP requests. Since edX uses Django for its backend services, the MFEs have to handle snake-cased <-> camelCase
+conversions manually. The 'axios-case-converter' package provides a middleware that would handle the conversion automatically. In the
+process of integrating the package, we thought it might be beneficial to
+establish a pattern for supporting middleware.
+
+Decision
+--------
+
+``getAuthenticatedHttpClient`` and ``getHttpClient`` take an ``options`` argument that includes configuration for the client.
+Only ``useCache`` is currently supported. We will introduce a new option ``middleware``, which will be a list of middleware
+that will be applied to the base axios instance. Each item in the list can either be a function or an array in the form of [function, options].
+We will apply the middleware and handle any exceptions that occur.
+
+It will look like this:
+    {
+        useCache: true,
+        middleware: [
+            axiosConvertCase,
+            [axiosRetry, { retries: 3 }],
+            ...
+        ]
+    }
+
+We will export any middleware functions that we support, i.e. ``applyCaseMiddleware`` from frontend-platform.
+The consumers will import the ones they want to use/add new ones and pass them in when creating a client.
```

Comment:
Since we are technically opening up the ability for consumers to create/apply their own middleware to the HTTP client, I wonder if it's worth including documentation such as "How do I create my own Axios client middleware?" as part of the README, etc. in `@edx/frontend-platform`, so consumers know this capability of extending the client is available to them.

For the purpose of this ADR, it might just be a "We will write documentation for how consumers can create their own middleware functions" for the time being.

Reply:
Idk if it should be part of the README, but perhaps on a how-to doc, and just external links to all the examples out there.